sklearn/tree/tree.py

```diff
@@ -152,6 +152,7 @@ def fit(self, X, y, sample_weight=None, check_input=True,
         random_state = check_random_state(self.random_state)
         if check_input:
             X = check_array(X, dtype=DTYPE, accept_sparse="csc")
+            y = check_array(y, accept_sparse='csc', ensure_2d=False, dtype=None)
```

Comment:
It should but converts them into csc form. Probably the csc format is optimized for the DecisionTree code hence.